packages/gatsby-source-contentful/src/download-with-retry.js

```diff
@@ -0,0 +1,40 @@
+const axios = require(`axios`)
+const rax = require(`retry-axios`)
+const { default: PQueue } = require(`p-queue`)
+
+const queue = new PQueue({
+  concurrency: 100,
+})
+
+export default async function downloadWithRetry(requestConfig, reporter) {
+  if (!requestConfig.url) {
+    throw new Error(`requestConfig.url is missing`)
+  }
+  const axiosInstance = axios.create()
+
+  axiosInstance.defaults.raxConfig = {
+    instance: axiosInstance,
+    onRetryAttempt: err => {
+      const cfg = rax.getConfig(err)
+      reporter.verbose(
+        `Retry attempt #${cfg.currentRetryAttempt} for ${requestConfig.url}`
+      )
+    },
+  }
+  rax.attach(axiosInstance)
```

Comment:
Do we want to create an axiosInstance for every download? Are connections shared?

Reply:
Thats what it did do before, but yeah, we could create one instance and reuse these

Reply:
If possible I would create it once. We could end up with 100 axios instances that hold memory, ... Maybe axios is smart enough but I don't know it well enough to vouch for it.